[WebAuthn] Avoid calling any ASC code if not present on the system

pascoej opened the pull request:
#### 243357fc57d2d69b36d7d350f4c3e6adc9be6839
<pre>
[WebAuthn] Avoid calling any ASC code if not present on the system
<a href="https://bugs.webkit.org/show_bug.cgi?id=259072">https://bugs.webkit.org/show_bug.cgi?id=259072</a>
rdar://112033763

Reviewed by Aditya Keerthi.

We need to avoid initializing any ASC code if we can&apos;t softlink it successfully.
In order to avoid error handling in struct initialization code, we first check
if the framework is available before proceeding with the request.

Canonical link: <a href="https://commits.webkit.org/265976@main">https://commits.webkit.org/265976@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/da9977d9c48b434e8b0feecfe03fd8ff23fac5e2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/12410 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/12745 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/13056 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/14152 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/11931 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/15171 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/12763 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/14604 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/12574 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/13325 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/10494 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/14575 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/10608 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/18337 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/11691 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/11404 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/14585 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/11895 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/9821 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/11127 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3059 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/15457 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/11750 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/UIProcess/WebAuthentication/WebAuthenticatorCoordinatorProxy.cpp:
```diff
@@ -79,6 +79,10 @@ void WebAuthenticatorCoordinatorProxy::handleRequest(WebAuthenticationRequestDat
         if (result) {
 #if HAVE(UNIFIED_ASC_AUTH_UI)
             if (!authenticatorManager.isMock() && !authenticatorManager.isVirtual()) {
+                if (!ascSuccessfullySoftlinked()) {
```
You should be able to use `isAuthenticationServicesCoreFrameworkAvailable()`, which is generated by the soft linking macros, instead of adding a new method.

achristensen07 commented on Source/WebKit/UIProcess/WebAuthentication/WebAuthenticatorCoordinatorProxy.cpp:
```diff
@@ -79,6 +79,10 @@ void WebAuthenticatorCoordinatorProxy::handleRequest(WebAuthenticationRequestDat
 #if HAVE(UNIFIED_ASC_AUTH_UI)
             if (!authenticatorManager.isMock() && !authenticatorManager.isVirtual()) {
+                if (!ascSuccessfullySoftlinked()) {
+                    handler({ }, (AuthenticatorAttachment)0, ExceptionData { NotSupportedError, "Not implemented."_s });
```
Let's not use a c-style cast, let's not cast an int to an enum.  Just have a default value of AuthenticatorAttachment::Platform, even though it's likely meaningless here.

pascoej committed:
[WebAuthn] Avoid calling any ASC code if not present on the system
https://bugs.webkit.org/show_bug.cgi?id=259072
rdar://112033763

Reviewed by Aditya Keerthi.

We need to avoid initializing any ASC code if we can't softlink it successfully.
In order to avoid error handling in struct initialization code, we first check
if the framework is available before proceeding with the request.

Canonical link: https://commits.webkit.org/265976@main
